lib/core.js

```diff
@@ -130,26 +120,27 @@ class Update {
     this.bitfield.setRange(batch.ancestors, batch.treeLength, false)
     this.batch.deleteBlockRange(bitfield.start, bitfield.start + bitfield.length)
 
-    const status = (batch.length > batch.ancestors) ? 0b0011 : 0b0010
-
     this.flushTreeBatch(batch)
-    this.coreUpdate({ status, bitfield, value: null, from })
+    this.updateContig(bitfield)
   }
 }
 
 class SessionState {
   constructor (core, storage, blocks, tree, bitfield, snapshotLength, name) {
     this.core = core
+    this.index = this.core.sessionStates.push(this) - 1
 
     this.storage = storage
     this.name = name
+    this.sessions = []
 
     this.mutex = new Mutex()
 
     this.blocks = blocks
     this.tree = tree
     this.bitfield = bitfield
     this.snapshotLength = snapshotLength
+    this.snapshotCompatLength = snapshotLength
```

Comment:
do we still need snapshotCompatLength separately?

Reply:
ya cause the normal one is used for the actual length and the compat is how much you can reuse (i think)